### ISSUE TITLE
OpenXR client patches for alvr 2023.01.30 (and alvr 2023.02.01) lags Quest2 & PICO 4

### ISSUE THREAD

**Issue by svsdval** [alvr-org/ALVR#1426]
## Description
alvr 2023.01.30 (openxr-client branch compiled from source)  & original nightly server:

1. if i set video resolution above 75% i got a lot of lags like on this video:
 https://user-images.githubusercontent.com/6585839/215772651-f42d9e93-03c2-4ef2-9de5-dda2d7d7ce8e.mp4
2. randomly pressed the left and right trigger (shots on video)

I tried to replace the compiled alvr server (openxr-client branch) with from original nightly 2023.01.30 and still have the same problem (so i think this is on openxr client side issue) 
any ideas ? or someone there is a working copy of the apk without this problem ?  on oculus quest 2 150-170% works fine.

## Environment
XFCE4

### Hardware
[hw-probe](https://linux-hardware.org/?probe=a2c87504d6)

### Installation
**ALVR Versions**:
[ALVR v20.0.0-dev06+nightly.2023.01.30](https://github.com/alvr-org/ALVR-nightly/releases/tag/v20.0.0-dev06%2Bnightly.2023.01.30)

**SteamVR Version**:
Built Jan 10, 2023 at 08:39
Version 1.25.3 (1673314780)

**Install Type**:
build from source & appimage

**OS Name and Version**
Debian GNU/Linux bookworm/sid

**Comment by deiteris** [comment]
It might be not a client issue, but the server. Could you show statistics graph? And did earlier versions work correctly for you?

**Comment by galister** [comment]
@svsdval are you using OpenVR Advanced Settings by any chance? I noticed similar rubber-banding when I tried to use it with ALVR on ArchLinux. Hopefully the upcoming pose-related changes will fix it.

**Comment by svsdval** [comment]
here is for oculus 3648x1920:
![oculus_3648x1920_1](https://user-images.githubusercontent.com/6585839/215925699-68b1154e-bc75-4fd3-a33d-2f03e409dadf.png)

here is for pico4 with same resolution: 3648x1920 in game stats:
![pico4_3648x1920_1](https://user-images.githubusercontent.com/6585839/215925706-77150f97-91fc-45a4-a07c-d484a22e33b6.png)
pico4 logs:
![pico4_3648x1920_logs](https://user-images.githubusercontent.com/6585839/215925710-d3785b04-9bf5-4658-a9a8-69199ff7bb0d.png)

here is a video for pico 4, with lags and random shots, in game fps lock on 72:
https://user-images.githubusercontent.com/6585839/215926785-89ff3043-b6b1-4d06-a301-7f4e834efc7d.mp4

**Comment by svsdval** [comment]
> @svsdval are you using OpenVR Advanced Settings by any chance? I noticed similar rubber-banding when I tried to use it with ALVR on ArchLinux. Hopefully the upcoming pose-related changes will fix it.

I don't think so, where can I find these settings or reset them?

**Comment by galister** [comment]
> > @svsdval are you using OpenVR Advanced Settings by any chance? I noticed similar rubber-banding when I tried to use it with ALVR on ArchLinux. Hopefully the upcoming pose-related changes will fix it.
> 
> I don't think so, where can I find these settings or reset them?

`OpenVR Advanced Settings` is a popular add-on can be installed via Steam. If you don't have it installed then please disregard.

**Comment by svsdval** [comment]
I don't have it installed =)

**Comment by deiteris** [comment]
> here is for oculus 3648x1920:
> 
> here is for pico4 with same resolution: 3648x1920 in game stats:
> 
> here is a video for pico 4, with lags and random shots, in game fps lock on 72: https://user-images.githubusercontent.com/6585839/215926785-89ff3043-b6b1-4d06-a301-7f4e834efc7d.mp4

This is not OK for your configuration. Can you check the beginning of the logs where it shows the information about the initialized encoder? It looks like it is using software encoding.

**Comment by svsdval** [comment]
SO, i tested on Linux & Windows, and found this setting : foveated_rendering ,if I turn on this option, the fps is like in windows, so the latency of the encoder drops to 8-14 ms, that's good, BUT! The lag in the video is still present. So I decided to try it on the Oculus, the issue is the same (tested on Windows & GNU/Linux), the video lags and is noticeable with and without this option, while I recorded how it “looks” the essence is that it looks twitchy in the helmet as it did on the video earlier, but when recording it appears as if the FPS is very low. 

For example:
Stats for  ALVR v20 0 0-dev06+nightly 2023 01 31_alvr_server_windows: enable_foveated_rendering = true:
![ALVR v20 0 0-dev06+nightly 2023 01 31_alvr_server_windows_enable_foveated_rendering_True](https://user-images.githubusercontent.com/6585839/216260533-fadc9402-bd87-4e6b-8e75-d4c2f03c7286.png)

Stats for  ALVR v20 0 0-dev06+nightly 2023 01 31_alvr_server_windows: enable_foveated_rendering = false:
![ALVR v20 0 0-dev06+nightly 2023 01 31_alvr_server_windows_enable_foveated_rendering_FALSE_stats](https://user-images.githubusercontent.com/6585839/216260525-7fc23d99-a7dc-49b6-842d-cec1705c1dca.png)

Stats for  ALVR v20 0 0-dev06+nightly 2023 01 31_alvr_server_windows: enable_foveated_rendering = false:
![ALVR v20 0 0-dev06+nightly 2023 01 31_alvr_server_windows_enable_foveated_rendering_FALSE](https://user-images.githubusercontent.com/6585839/216260520-6913c82c-4413-42c6-b7bb-4ea23be5fb77.png)

Settings for  2023 01 31:
![ALVR v20 0 0-dev06+nightly 2023 01 31_alvr_server_windows_enable_foveated_rendering_FALSE_video_settings](https://user-images.githubusercontent.com/6585839/216260530-9ca4abbe-caae-4cee-bffe-2289e19d341d.png)

Settings for  2023 02 01:
![ALVR v20 0 0-dev07+nightly 2023 02 01_alvr_server_windows_enable_foveated_rendering_FALSE_settings](https://user-images.githubusercontent.com/6585839/216260538-d113d3a7-8eac-433b-9878-b22be050b8be.png)

Here is video recorded from 2023 01 31 with DISABLED foveated_rendering:
https://user-images.githubusercontent.com/6585839/216261356-ae5d5c9e-1aa7-45a2-bc48-135bcc130150.mp4

Here is video recorded from 2023 02 01 with DISABLED foveated_rendering:
https://user-images.githubusercontent.com/6585839/216261364-bd334076-66ee-463f-97a8-4cc76c2e4da8.mp4

**Comment by svsdval** [comment]
Here is a GNU/Linux Stats for foveated_rendering = True
![v20 0 0-dev07+nightly 2023 02 01_alvr_server_windows_enable_foveated_rendering_True_settings png](https://user-images.githubusercontent.com/6585839/216266142-871fa986-f282-4d9b-94be-c09508aa08eb.png)
Here is a GNU/Linux settings for foveated_rendering = True
![v20 0 0-dev07+nightly 2023 02 01_alvr_server_windows_enable_foveated_rendering_True_stats1](https://user-images.githubusercontent.com/6585839/216266148-fd54883c-f7af-4711-8dcb-89ffc566bcf4.png)
Here is a GNU/Linux video with lags for foveated_rendering = True
https://user-images.githubusercontent.com/6585839/216266397-41f401c1-ccbd-47a1-ac17-3922dc695c21.mp4

Here is a GNU/Linux video with lags for foveated_rendering = False:
https://user-images.githubusercontent.com/6585839/216267063-66ad0e07-61a2-4ed6-9ebc-f064cba8a315.mp4


So this is issue OpenXR patches

**Comment by galister** [comment]
For Pico try setting both HMD and ALVR to 90hz

**Comment by svsdval** [comment]
> For Pico try setting both HMD and ALVR to 90hz
@galister, did not help

**Comment by svsdval** [comment]
Yes I checked with ALVR server 2023-01-31 for clients 2023-02-01 and 2023-02-02 for Oculus and Pico 4 same problem (twitchy if i disable the foveated_rendering) so client side issue

**Comment by zarik5** [comment]
Are you sure this is not a Linux issue? Make sure to disable async reprojection in the Extra tab.

**Comment by Kirottu** [comment]
Running a game that is light-er (like Synth Riders), which does not drop frames runs fine. But anything that does drop frames suffers from this in my testing, you can use the SteamVR perf graph inside the headset to see this behavior (Orange/Purple indicates dropped frames depending on if you have legacy reprojection enabled).

**Comment by svsdval** [comment]
The reprojection is disabled on both Windows 10 (515 drivers) and Linux (520 drivers). To determine where the problem is on the server side or client side, I ran 2023-01-31 (server) with the Oculus client (2023-01-31) and no problems, then I ran the Oculus Andorid client (2023-02-01/02) and got this problem. So this is problem in Pico4 patches...

**Comment by deiteris** [comment]
Possibly fixed with https://github.com/alvr-org/ALVR/pull/1442. Could you try latest nightly?

**Comment by svsdval** [comment]
Almost fixed.
1. It seems that the FPS has fallen at times, although the statistics say that everything is OK fps = 71-75-69...
2. Now the right eye looks/feels smooth, but the left eye twitches (not allways, and it feels like a hit on the eye, it feels like the picture became to static for several tens of milliseconds, then a sharp hit on the eye when the picture changes), I tried to record a video, but on the recording created using HMD, this is not visible (it looks like the picture is taken from the right eye). It seems that the fps of the left eye is lower than the fps of the right one O_o, maybe the eyes are not synchronized? I tested with and without foveated rendering.
